### ISSUE TITLE
Document : clicking on the link conduct to an empty page

### ISSUE THREAD

**Issue by tanchon** [chamilo/chamilo-lms#3540]
Hi,

There is a little problem with some documents that the "showinframe" cannot read : for example, with a msWord document, when i click on the link of the document, the download works fine... but a new page open, and it's empty.

I think that in that case, the click should only have one consequence => to download the document : the browser should stay on the list of documents.

**Screenshots**
![Capture](https://user-images.githubusercontent.com/24507698/94335217-514a5200-ffda-11ea-927d-82509b33c3ba.JPG)

![Capture2](https://user-images.githubusercontent.com/24507698/94335224-560f0600-ffda-11ea-8432-015a24455e20.JPG)

**Server
Chamilo 1.11.12. Php 7.3

**Additional context**
When we use a lot of documents, this problem is boring.

Thanks.

Tanchon

**Comment by ywarnier** [comment]
I agree but we will probably not be able to integrate this change in 1.11.14.

**Comment by tanchon** [comment]
Hi @ywarnier 

OK.  I found a way to avoid this issue. I applied this, some weeks ago, and it seems to be ok (I did not have any problem with this little adjustment).

![Capture](https://user-images.githubusercontent.com/24507698/100089395-e7bbb800-2e51-11eb-9fa9-53f2af0f78a0.JPG)

After that, when i click on a link (.doc, xls...) the file is downloaded without a new blank page opening. My students are happy with that !

If you think it could generate any issue, could you tell me please ?

Tanchon.

**Comment by ywarnier** [comment]
I think the issue is in the documents listing: the document name should not have a link to "view" the document if it's clear the browser will not be able to read it and display it on its own.

**Comment by ywarnier** [comment]
Testing procedure:
- [ ] login as a teacher
- [ ] create a new course
- [ ] go to the documents tool
- [ ] upload several documents: at least one PDF, one XLSX, one DOCX, one HTML, one PPTX and one ODT
- [ ] make sure the documents are visible to students
- [ ] return to course homepage
- [ ] make sure the documents tool is visible to students
- [ ] enter the "users" tool
- [ ] subscribe a student to the course
- [ ] logout
- [ ] login as the student
- [ ] enter the course
- [ ] enter the documents tool
- [ ] click the name of each document and check they are opened or downloaded (whichever one) but that you are not prevented from seeing the document, or that a new blank page does not open and remain there while the document is downloaded to your computer

If all of this checks out and you have no significant observation, this issue can be closed.

**Comment by klineur-M** [comment]
@ywarnier  **FAILED**

Steps 1 > 13 : ok
Step 14 :   .docx --  .xlsx  --  .pptx --  .odt are working fine ( downloaded & opened : ok )
**BUT**  .html  &  .pdf are not downloaded the way the 4 others effectively are.

**Comment by christianbeeznest** [comment]
It is improved and fixed with this commit https://github.com/chamilo/chamilo-lms/pull/6533/commits/f39af04e9be76a601a3f51383674b0b030f34393

Thanks for confirmation.

**Comment by klineur-M** [comment]
@ywarnier : testing procedure applied. _( FF and Opera browsers ; on Opera I had to logout - login again to get it working )_ **Works fine**, issue might be closed
